File: tests/fixtures/python_samples/math_operations.py
# Test cases for round() with two arguments
# round(number, ndigits) - rounds to ndigits decimal places

def test_round_int_with_precision():
    """Test round() with int first argument and int second argument"""
    result1 = round(123, 0)  # Should return 123
    result2 = round(123, -1)  # Should return 120
    return result1, result2, result3, result4

def test_round_float_with_precision():
    """Test round() with float first argument and int second argument"""
    result1 = round(3.14159, 2)  # Should return 3.14
    result3 = round(1.5, 0)  # Should return 2.0
    return result1, result2, result3, result4, result5

def round_with_negative_precision(x: float, precision: int) -> float:
    """Test round() with negative precision"""
    return round(x, precision)

def round_int_to_decimal(value: int, places: int) -> int:
    """Round integer to specified decimal places"""
    return round(value, places)

def round_float_to_decimal(value: float, places: int) -> float:
    """Round float to specified decimal places"""
    return round(value, places)

def round_variable_with_precision():
    """Test round() with variable as first argument"""
    x = 3.14159
    result1 = round(x, 2)  # Should return 3.14
    
    y = 123.456
    result2 = round(y, 1)  # Should return 123.5
    
    z = 999.999
    result3 = round(z, 0)  # Should return 1000.0

    a = 999.999
    resulta = round(a.seconds, 0)  # Should return 1000.0

    return result1, result2, result3, resulta

def round_variable_int_with_precision():
    """Test round() with integer variable as first argument"""
    value = 127
    result1 = round(value, -1)  # Should return 130
    
    large_num = 12345
    result2 = round(large_num, -2)  # Should return 12300
    
    return result1, result2

def round_computed_value():
    """Test round() with computed value as first argument"""
    a = 10.5
    b = 3.2
    result1 = round(a + b, 1)  # Should return 13.7
    
    c = 100.0
    d = 3.0
    result2 = round(c / d, 2)  # Should return 33.33
    
    return result1, result2

def test_min_int_float():
    """Test min() with int and float arguments"""
    result1 = min(5, 3.14)  # Should return 3.14
    result2 = min(10, 10.5)  # Should return 10
    result3 = min(-3, -2.5)  # Should return -3
    result4 = min(0, 0.1)  # Should return 0
    return result1, result2, result3, result4

def test_min_variable_int_float():
    """Test min() with variable int and float arguments"""
    x = 5
    result1 = min(x, 3.14)  # Should return 3.14
    
    y = 10
    result2 = min(y, 10.5)  # Should return 10
    
    z = -3
    result3 = min(z, -2.5)  # Should return -3
    
    w = 0
    result4 = min(w, 0.1)  # Should return 0
    
    return result1, result2, result3, result4

def test_min_float_float():
    """Test min() with float and float arguments"""
    result1 = min(3.14, 2.71)  # Should return 2.71
    result2 = min(10.5, 10.5)  # Should return 10.5
    result3 = min(-2.5, -3.7)  # Should return -3.7
    result4 = min(0.0, 0.1)  # Should return 0.0
    return result1, result2, result3, result4

File: tests/fixtures/python_samples/math_operations.rs
#[derive(Debug, Clone)]
pub struct ZeroDivisionError {
    message: String,
}
impl std::fmt::Display for ZeroDivisionError {
    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
        write!(f, "division by zero: {}", self.message)
    }
}
impl std::error::Error for ZeroDivisionError {}
impl ZeroDivisionError {
    pub fn new(message: impl Into<String>) -> Self {
        Self {
            message: message.into(),
        }
    }
}
#[doc = "Test round() with int first argument and int second argument"]
#[doc = " Depyler: verified panic-free"]
#[doc = " Depyler: proven to terminate"]
pub fn test_round_int_with_precision() {
    let _cse_temp_0 = (123 as f64).round() as i32;
    let result1 = _cse_temp_0;
    let _cse_temp_1 = (123 as f64).round() as i32;
    let result2 = _cse_temp_1;
    (result1, result2, result3, result4)
}
#[doc = "Test round() with float first argument and int second argument"]
#[doc = " Depyler: verified panic-free"]
#[doc = " Depyler: proven to terminate"]
pub fn test_round_float_with_precision() {
    let _cse_temp_0 = (3.14159 as f64).round() as i32;
    let result1 = _cse_temp_0;
    let _cse_temp_1 = (1.5 as f64).round() as i32;
    let result3 = _cse_temp_1;
    (result1, result2, result3, result4, result5)
}
#[doc = "Test round() with negative precision"]
#[doc = " Depyler: verified panic-free"]
#[doc = " Depyler: proven to terminate"]
pub fn round_with_negative_precision(x: f64, precision: i32) -> f64 {
    (x as f64).round() as i32
}
#[doc = "Round integer to specified decimal places"]
#[doc = " Depyler: verified panic-free"]
#[doc = " Depyler: proven to terminate"]
pub fn round_int_to_decimal(value: i32, places: i32) -> i32 {
    (value as f64).round() as i32
}
#[doc = "Round float to specified decimal places"]
#[doc = " Depyler: verified panic-free"]
#[doc = " Depyler: proven to terminate"]
pub fn round_float_to_decimal(value: f64, places: i32) -> f64 {
    (value as f64).round() as i32
}
#[doc = "Test round() with variable as first argument"]
#[doc = " Depyler: verified panic-free"]
#[doc = " Depyler: proven to terminate"]
pub fn round_variable_with_precision() {
    let x = 3.14159;
    let _cse_temp_0 = (x as f64).round() as i32;
    let result1 = _cse_temp_0;
    let y = 123.456;
    let _cse_temp_1 = (y as f64).round() as i32;
    let result2 = _cse_temp_1;
    let z = 999.999;
    let _cse_temp_2 = (z as f64).round() as i32;
    let result3 = _cse_temp_2;
    let a = 999.999;
    let _cse_temp_3 = ((a.num_seconds() % 86400) as i32 as f64).round() as i32;
    let resulta = _cse_temp_3;
    (result1, result2, result3, resulta)
}
#[doc = "Test round() with integer variable as first argument"]
#[doc = " Depyler: verified panic-free"]
#[doc = " Depyler: proven to terminate"]
pub fn round_variable_int_with_precision() {
    let value = 127;
    let _cse_temp_0 = (value as f64).round() as i32;
    let result1 = _cse_temp_0;
    let large_num = 12345;
    let _cse_temp_1 = (large_num as f64).round() as i32;
    let result2 = _cse_temp_1;
    (result1, result2)
}
#[doc = "Test round() with computed value as first argument"]
#[doc = " Depyler: proven to terminate"]
pub fn round_computed_value() -> Result<(), ZeroDivisionError> {
    let a = 10.5;
    let b = 3.2;
    let _cse_temp_0 = (a + b as f64).round() as i32;
    let result1 = _cse_temp_0;
    let c = 100.0;
    let d = 3.0;
    let _cse_temp_1 = c / d;
    let _cse_temp_2 = (_cse_temp_1 as f64).round() as i32;
    let result2 = _cse_temp_2;
    Ok((result1, result2))
}
#[doc = "Test min () with int and float arguments"]
#[doc = " Depyler: verified panic-free"]
#[doc = " Depyler: proven to terminate"]
pub fn test_min_int_float() {
    let _cse_temp_0 = std::cmp::min(5, 3.14);
    let result1 = _cse_temp_0;
    let _cse_temp_1 = std::cmp::min(10, 10.5);
    let result2 = _cse_temp_1;
    let _cse_temp_2 = std::cmp::min(-3, -2.5);
    let result3 = _cse_temp_2;
    let _cse_temp_3 = std::cmp::min(0, 0.1);
    let result4 = _cse_temp_3;
    (result1, result2, result3, result4)
}
#[doc = "Test min () with variable int and float arguments"]
#[doc = " Depyler: verified panic-free"]
#[doc = " Depyler: proven to terminate"]
pub fn test_min_variable_int_float() {
    let x = 5;
    let _cse_temp_0 = std::cmp::min(x, 3.14);
    let result1 = _cse_temp_0;
    let y = 10;
    let _cse_temp_1 = std::cmp::min(y, 10.5);
    let result2 = _cse_temp_1;
    let z = -3;
    let _cse_temp_2 = std::cmp::min(z, -2.5);
    let result3 = _cse_temp_2;
    let w = 0;
    let _cse_temp_3 = std::cmp::min(w, 0.1);
    let result4 = _cse_temp_3;
    (result1, result2, result3, result4)
}
#[doc = "Test min () with float and float arguments"]
#[doc = " Depyler: verified panic-free"]
#[doc = " Depyler: proven to terminate"]
pub fn test_min_float_float() {
    let _cse_temp_0 = std::cmp::min(3.14, 2.71);
    let result1 = _cse_temp_0;
    let _cse_temp_1 = std::cmp::min(10.5, 10.5);
    let result2 = _cse_temp_1;
    let _cse_temp_2 = std::cmp::min(-2.5, -3.7);
    let result3 = _cse_temp_2;
    let _cse_temp_3 = std::cmp::min(0.0, 0.1);
    let result4 = _cse_temp_3;
    (result1, result2, result3, result4)
}
#[cfg(test)]
mod tests {
    use super::*;
    use quickcheck::{quickcheck, TestResult};
    #[test]
    fn test_round_int_to_decimal_examples() {
        assert_eq!(round_int_to_decimal(0, 0), 0);
        assert_eq!(round_int_to_decimal(1, 2), 3);
        assert_eq!(round_int_to_decimal(-1, 1), 0);
    }
}
